creusot/src/analysis/not_final_places.rs

```diff
@@ -230,6 +224,11 @@ impl<'tcx> AnalysisDomain<'tcx> for NotFinalPlaces<'tcx> {
     fn initialize_start_block(&self, _: &mir::Body<'tcx>, _: &mut Self::Domain) {}
 }
 
+// The NotFinalPlaces analysis computes, for each location, places which either:
+// - do not contain a mutable borrow deref and may be moved or borrowed in the future
```

Comment:
We should have a precise example of this, or even a test added to `should_fail/final_borrows.rs`
Something like 
```rust
// `bor` is a place that does not contain a mutable deref, but it is move after it has been reborrowed.
pub fn move_place_without_deref<T>(bor: &mut T) {
    let b1 = &mut *bor;
    proof_assert!(b1 == bor);
    f(bor);
}
#[requires(true)]
#[ensures(true)]
fn f<T>(x: &mut T) {}
```

Reply:
Too late, sorry.

Feel free to add a test in your future PR. But note that in your example, `f(bor)` is actually a reborrow of `bor`, and not a move. To force a move, the simplest is to wrap the borrow under a `Box`.